docs/src/adoc/index.adoc

```diff
@@ -2409,18 +2410,35 @@ ClasspathSqlLocator.findSqlOnClasspath(com.foo.BarDao.class, "query");
 ClasspathSqlLocator.findSqlOnClasspath("com.foo.BarDao.query");
 ----
 
-=== StatementContext
+== Configuration
 
-The link:{jdbidocs}/core/statement/StatementContext.html[StatementContext] class
-is a carrier for various state related to the creation and execution of statements
-that is not appropriate to hold on the *Query* or other particular statement class
-itself.  Among other things, it holds open *JDBC* resources, processed SQL statements,
-and accumulated bindings.  It is exposed to implementations of most user extension points,
-for example *RowMapper, *ColumnMapper*s, or *CollectorFactory*.
+`Jdbi` aims to be useful out of the box with minimal configuration.  Sometimes
+you need to change default behavior, or add in extensions to handle additional
+database types.  Each piece of core or extension that wishes to participate in
+configuration defines a configuration class, for example the `SqlStatements`
+class stores Handle related configuration.  Then, on any `Configurable` context
+(like a `Jdbi` or `Handle`) you can change configuration in a type safe way:
 
-The *StatementContext* itself is not intended to be extended and generally extensions
-should not need to mutate the context.  Please read the JavaDoc for more information
-on advanced usage.
+[source,java]
+----
+jdbi.getConfig(SqlStatements.class).setUnusedBindingAllowed(true);
+jdbi.getConfig(Arguments.class).register(new MyTypeArgumentFactory());
+
+// Or, if you have a bunch of work to do:
+jdbi.configure(RowMappers.class, rm -> {
+    rm.register(new TypeARowMapperFactory();
+    rm.register(new TypeBRowMapperFactory();
+});
+----
+
+Generally, you should finalize all configuration changes before interacting with
+the database.
+
+When a new context is created, it inherits a copy of the parent context
+configuration at the time of creation.  So a `Handle` initializes its
+configuration from the creating `Jdbi`, but changes never propagate back up.
+
+See <<JdbiConfig>> for the gory implementation details.
```

Comment:
hah

Reply:
This may be a stupid question but what is the meaning of _gory_ here? Did you mean gory like in the gore movies? Can we have this line changed to:

```plain
See <<JdbiConfig>> for more advanced implementation details.
```

Reply:
Sorry, that's slang for essentially the same thing.  I guess we can be more proper :P